### PR TITLE
xtest 6017: Test info.dataSize in Persistent objects

### PR DESCRIPTION
**Pull request by lackan**
Create a persistent object and write some data into it, get the info.dataSize and then close it, and reopen it
to get the new info.dataSize of this reopen's object, these two dataSizes should be equal.

Relate to OP-TEE/optee_os#1217

Signed-off-by: Guanchao Liang <liang.guanchao@linaro.org>

### CODE REVIEW

**Comment by jbech-linaro** [comment]
`Reviewed-by: Joakim Bech <joakim.bech@linaro.org>`

**Comment by jenswi-linaro** [comment]
The test itself is good, but I'd prefer if you added a `TA_STORAGE_CMD_GET_OBJECT_INFO` to the TA and did the testing in the test case instead of inside the ta.

**Comment by lackan** [comment]
@jenswi-linaro OK, I will change as your comment.

File: ta/storage/storage.c [hunk 1]
```diff
@@ -485,3 +485,62 @@ TEE_Result ta_storage_cmd_reset_obj(uint32_t param_types, TEE_Param params[4])
...
 }
+
+TEE_Result ta_storage_cmd_info_in_persistent(uint32_t param_types,
+					    TEE_Param params[4])
+{
+	TEE_Result res;
+	TEE_ObjectHandle object;
+	TEE_ObjectInfo info1;
+	TEE_ObjectInfo info2;
+	char write_data[50];
```
Comment (igoropaniuk) [inline ta/storage/storage.c]: minor: magic "50" for data length, maybe it's better to have one constant somewhere for it?

File: ta/storage/storage.c [hunk 2]
```diff
@@ -485,3 +485,62 @@ TEE_Result ta_storage_cmd_reset_obj(uint32_t param_types, TEE_Param params[4])
...
+	if (res != TEE_SUCCESS) {
+		EMSG("Failed to open persistent object: 0x%x", res);
+		return res;
+	}
+
+	TEE_GetObjectInfo(object, &info2);
+	DMSG("Reopen object data size : %u and position : %u",
+		info2.dataSize, info2.dataPosition);
+
+	if (info1.dataSize != info2.dataSize)
```
Comment (igoropaniuk) [inline ta/storage/storage.c]: What about also to compare with primary write_data_len ?

**Comment by jforissier** [comment]
`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Comment by jenswi-linaro** [comment]
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by lackan** [comment]
Have added Review-by tags and rebased onto current master branch.

**Pull request merged**